taskvine/src/worker/vine_cache.c

```diff
@@ -159,7 +159,9 @@ static void vine_cache_kill(struct vine_cache *c, struct vine_cache_file *f, con
 	while (f->status == VINE_CACHE_STATUS_PROCESSING) {
 		debug(D_VINE, "cache: killing pending transfer process %d...", f->pid);
 		kill(f->pid, SIGKILL);
-		vine_cache_wait_for_file(c, f, cachename, manager);
+		if (manager) {
```

Comment:
vine_cache_wait_for_file needs to be called unconditionally.
It does a variety of things, in addition to sending updates to the manager.